corehq/apps/cloudcare/views.py

```diff
@@ -210,6 +212,20 @@ def _default_lang():
             "integrations": integration_contexts(domain),
             "has_geocoder_privs": has_geocoder_privs(domain),
         }
+
+        langs_by_code = {}
+
+        def get_lang_name(lang_code):
+            if not langs_by_code.get(lang_code):
+                langs_by_code[lang_code] = get_name(lang_code)
+            return langs_by_code[lang_code]
+
+        for app in apps:
+            lang_codes = app.get("langs", [])
+            lang_code_name_mapping = {code: get_lang_name(code) for code in lang_codes}
+            app["lang_code_name_mapping"] = lang_code_name_mapping
+            app.pop("langs")
```

Comment:
why do we need to remove langs from app?
If we do, you could simply pop is to set variable lang_codes.

Reply:
Isn't removing langs going to break code like [this](https://github.com/dimagi/commcare-hq/blob/586c9a8e2b67d2d470233a1133f39080a3f64a76/corehq/apps/cloudcare/static/cloudcare/js/formplayer/layout/views/settings.js#L37)?

Reply:
Thanks for catching that @orangejenny